### PR TITLE
adding "grab n slide" Musette into marble

### PR DESCRIPTION
**Pull request by davidseguin**

### CODE REVIEW

File: dist/marble.js.LICENSE.txt [hunk 1]
```diff
@@ -45,7 +45,1201 @@
  * @license MIT
  */
 
+/*! ../../../global/settings */
```
Comment (stephenhmarsh) [inline dist/marble.js.LICENSE.txt:48]: I'm not sure what all this cruft is? 🤔  Could you just delete this file and rebuild dist?

File: dist/marble.js [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
...-playing"),this.analyticsSender.sendCustomEvent({event:"UniversalAudioPlayer:trackPlayed",trackId:this.currentTrack.id,trackTitle:this.currentTrack.title})}handlePause(){this.wrapperEl.classList.remove("is-playing")}quickSeekBack(e){e.preventDefault();const t=Math.max(0,this.audioEl.currentTime-this.seekHelperDuration);this.audioEl.currentTime=t}quickSeekForward(e){e.preventDefault();const t=Math.min(this.audioEl.duration,this.audioEl.currentTime+this.seekHelperDuration);this.audioEl.currentTime=t}handleTranscriptToggle(e){e.preventDefault(),this.transcriptSection.classList.contains("transcript-is-open")?(this.transcriptToggleText.innerHTML=this.quoteExpanderDefaultText,this.transcriptSection.classList.remove("transcript-is-open")):(this.transcriptSection.classList.add("transcript-is-open"),this.transcriptToggleText.innerHTML="Hide Transcript",this.analyticsSender.sendCustomEvent({event:"UniversalAudioPlayer:transcriptOpened",trackId:this.currentTrack.id,trackTitle:this.currentTrack.title}))}}},"./src/components/audio-player/cover-image-template.js":(e,t,s)=>{"use strict";s.r(t);var n=s("./node_modules/underscore/modules/index-all.js");t.default=e=>{let t=String.raw;return e?t`<img
```
Comment (stephenhmarsh) [inline dist/marble.js:2]: Hm, not sure how but looks like a broken build?

File: package-lock.json [hunk 3]
```diff
@@ -19664,7 +19664,239 @@
 				"validate-npm-package-name",
 				"which",
 				"worker-farm",
-				"write-file-atomic"
+				"write-file-atomic",
+				"agent-base",
```
Comment (stephenhmarsh) [inline package-lock.json:19668]: looks like maybe you had a bad merge?

File: src/components/musette/musette.stories.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
+import { html } from ".storybook/helpers";
+import { withKnobs } from "@storybook/addon-knobs";
+import lesMusettes from "./musette.js";
+import { useEffect } from "@storybook/client-api";
+
+export default { title: "Carousel/Musette", decorators: [withKnobs] };
```
Comment (stephenhmarsh) [inline src/components/musette/musette.stories.js:6]: the knobs addon has been buggy. better to just remove it, especially if you don't have any.  we're now slowly migrating to controls/args.

File: src/components/musette/musette.stories.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+const Musette = () => {
+	useEffect(() => { lesMusettes(); });
+	return html`
+	<div class="la-musette la-musette-no-scrollbar js-la-musette">
+		<div class="musette-fold">
+			<a href="http://www.metmuseum.org/art/collection/search/771396">
+				<img src="https://collectionapi.metmuseum.org/api/collection/v1/iiif/503578/1358034/main-image">
```
Comment (stephenhmarsh) [inline src/components/musette/musette.stories.js:14]: probably better if you can change the `src` for all of these to one of our storybook image helpers, so we're not depending on non-local assets in test environments for snapshots. not a blocker but its preferred.

File: src/components/musette/musette.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+		laMusette.parentNode.insertBefore(musetteWrapper, laMusette);
+
+		// put elements into wrapper
+		musetteWrapper.appendChild(laMusette);
+		musetteWrapper.appendChild(leftButton);
+		musetteWrapper.appendChild(rightButton);
+
+		// get dimensions, used exclusively for buttons
+		// imported from old Madaveousel and maybe more specs than we need
+		const getMusetteSpecs = (laMusette) => {
```
Comment (stephenhmarsh) [inline src/components/musette/musette.js:34]: i think @JosephCurley wanted to use https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API ?

File: src/components/musette/musette.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+			var thisMusette = getMusetteSpecs(laMusette);
+			var scrollToPosition = thisMusette.positionLeft - thisMusette.viewportWidth;
+			laMusette.scrollTo({left: scrollToPosition, behavior: "smooth"});
+			checkButtonStatus(laMusette);
+		};
+
+		// listen for scroll, so we can check for buttons when it"s finished
+		var isScrollingNow;
+    
+		laMusette.addEventListener("scroll", () => {
```
Comment (stephenhmarsh) [inline src/components/musette/musette.js:86]: should we throttle or debounce this, for performance?

File: src/components/musette/musette.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+		});
+		
+		laMusette.addEventListener("mouseleave", () => {
+			mouseIsDown = false;
+		});
+		
+		laMusette.addEventListener("mouseup", () => {
+			const musettteLinks = laMusette.querySelectorAll("a");
+			if(mouseIsBeingDragged){
+				for(let i = 0; i<musettteLinks.length; i++) {
```
Comment (stephenhmarsh) [inline src/components/musette/musette.js:122]: refactor to `.forEach` ?

**Review by stephenhmarsh — CHANGES_REQUESTED** [review]

File: src/components/musette/musette.scss [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+	&::-webkit-scrollbar {
+		display: none;
+	}
+}
+
+//navigation
+.musette-move-left,
+.musette-move-right {
+	background: no-repeat center/20px url("https://www.metmuseum.org/Rodan/dist/svg/banner-link-back.svg");
```
Comment (stephenhmarsh) [inline src/components/musette/musette.scss:37]: I think we shouldn't depend on a file in Rodan here. Probably best to inline it as a data-uri svg. ? What do you think @JosephCurley

**Review by JosephCurley — APPROVED** [review]
This is looking good - I can imagine a few places to take it, but Im cool with merging it in whenever

**Review by stephenhmarsh — APPROVED** [review]
great job, thanks @davidseguin

**Pull request merged**